src/components/visualizations/chart/highcharts/plugins/autohideLabels/autohideBarLabels.ts

```diff
@@ -0,0 +1,91 @@
+// (C) 2007-2018 GoodData Corporation
+import sortBy = require('lodash/sortBy');
+
+import {
+    isStacked,
+    getVisibleSeries,
+    getDataPoints,
+    showDataLabels,
+    hideDataLabels,
+    showDataLabel,
+    hideDataLabel,
+    hasDataLabel,
+    toNeighbors,
+    isIntersecting,
+    getDataLabelAttributes,
+    getShapeAttributes
+} from '../../helpers';
+
+const toggleStackedChartLabels = (visiblePoints: any) => {
+    const intersectionFound = visiblePoints
+        .filter(hasDataLabel)
+        .some((point: any) => {
+            const { dataLabel, shapeArgs } = point;
+
+            if (dataLabel && shapeArgs) {
+                const dataLabelAttr = getDataLabelAttributes(point);
+                const shapeAttr = getShapeAttributes(point);
+                return dataLabelAttr.height + (2 * dataLabel.padding) > shapeAttr.height;
+            }
+            return false;
+        });
+
+    if (intersectionFound) {
+        hideDataLabels(visiblePoints);
+    } else {
+        visiblePoints.filter(hasDataLabel).forEach((point: any) => {
+            const { dataLabel, shapeArgs } = point;
+            if (dataLabel && shapeArgs) {
+                const dataLabelAttr = getDataLabelAttributes(point);
+                const shapeAttr = getShapeAttributes(point);
+                const labelWidth = dataLabelAttr.width + (2 * dataLabel.padding);
+                const foundIntersection = labelWidth > shapeAttr.width;
+                return foundIntersection ? hideDataLabel(point) : showDataLabel(point);
+            }
+            return null;
+        });
+    }
+};
+
+const toggleNonStackedChartLabels = (points: any, shouldCheckShapeIntersection: any = false) => {
```

Comment:
I guess in this case we can be sure that `shouldCheckShapeIntersection` is of type `boolean`.